Index: llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/clang/lib/Driver/ToolChains/WebAssembly.cpp
===================================================================
--- llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b.orig/clang/lib/Driver/ToolChains/WebAssembly.cpp
+++ llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/clang/lib/Driver/ToolChains/WebAssembly.cpp
@@ -21,6 +21,11 @@
 #include "llvm/Support/Path.h"
 #include "llvm/Support/VirtualFileSystem.h"
 
+#include "llvm/Config/llvm-config.h" // for LLVM_VERSION_MAJOR
+
+#define TOSTR2(X) #X
+#define TOSTR(X) TOSTR2(X)
+
 using namespace clang::driver;
 using namespace clang::driver::tools;
 using namespace clang::driver::toolchains;
@@ -257,7 +262,7 @@ WebAssembly::WebAssembly(const Driver &D
 const char *WebAssembly::getDefaultLinker() const {
   if (TargetBuildsComponents(getTriple()))
     return "wasm-component-ld";
-  return "wasm-ld";
+  return "wasm-ld-" TOSTR(LLVM_VERSION_MAJOR);
 }
 
 bool WebAssembly::IsMathErrnoDefault() const { return false; }
