Index: llvm-toolchain-snapshot_17~++20230517015648+0b6264738f3d/lldb/tools/argdumper/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_17~++20230517015648+0b6264738f3d.orig/lldb/tools/argdumper/CMakeLists.txt
+++ llvm-toolchain-snapshot_17~++20230517015648+0b6264738f3d/lldb/tools/argdumper/CMakeLists.txt
@@ -6,3 +6,6 @@ add_lldb_tool(lldb-argdumper ADD_TO_FRAM
 )
 
 add_dependencies(liblldb lldb-argdumper)
+
+install(TARGETS lldb-argdumper
+  RUNTIME DESTINATION bin)
Index: llvm-toolchain-snapshot_17~++20230517015648+0b6264738f3d/lldb/tools/lldb-server/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_17~++20230517015648+0b6264738f3d.orig/lldb/tools/lldb-server/CMakeLists.txt
+++ llvm-toolchain-snapshot_17~++20230517015648+0b6264738f3d/lldb/tools/lldb-server/CMakeLists.txt
@@ -68,3 +68,7 @@ add_dependencies(lldb-server
 )
 target_include_directories(lldb-server PRIVATE "${LLDB_SOURCE_DIR}/source")
 target_link_libraries(lldb-server PRIVATE ${LLDB_SYSTEM_LIBS})
+
+install(TARGETS lldb-server
+  RUNTIME DESTINATION bin)
+
