Index: llvm-toolchain-snapshot_20~++20241014014336+6a98c4a16025/llvm/lib/TargetParser/ARMTargetParser.cpp
===================================================================
--- llvm-toolchain-snapshot_20~++20241014014336+6a98c4a16025.orig/llvm/lib/TargetParser/ARMTargetParser.cpp
+++ llvm-toolchain-snapshot_20~++20241014014336+6a98c4a16025/llvm/lib/TargetParser/ARMTargetParser.cpp
@@ -641,9 +641,9 @@ StringRef ARM::getARMCPUForArch(const ll
     case llvm::Triple::GNUEABIHF:
     case llvm::Triple::GNUEABIHFT64:
     case llvm::Triple::MuslEABIHF:
-      return "cortex-a7";
+      return "cortex-a8";
     default:
-      return "arm7tdmi";
+      return "arm926ej-s";
     }
   }
 
