Index: llvm-toolchain-snapshot_19~++20240711013116+0ff13f72c974/openmp/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_19~++20240711013116+0ff13f72c974.orig/openmp/CMakeLists.txt
+++ llvm-toolchain-snapshot_19~++20240711013116+0ff13f72c974/openmp/CMakeLists.txt
@@ -51,13 +51,13 @@ else()
   set(OPENMP_ENABLE_WERROR ${LLVM_ENABLE_WERROR})
 
   # When building in tree we install the runtime according to the LLVM settings.
-  if(LLVM_ENABLE_PER_TARGET_RUNTIME_DIR AND NOT APPLE)
-    set(OPENMP_INSTALL_LIBDIR lib${LLVM_LIBDIR_SUFFIX}/${LLVM_DEFAULT_TARGET_TRIPLE} CACHE STRING
-        "Path where built openmp libraries should be installed.")
-  else()
+#  if(LLVM_ENABLE_PER_TARGET_RUNTIME_DIR AND NOT APPLE)
+#    set(OPENMP_INSTALL_LIBDIR lib${LLVM_LIBDIR_SUFFIX}/${LLVM_DEFAULT_TARGET_TRIPLE} CACHE STRING
+#        "Path where built openmp libraries should be installed.")
+#  else()
     set(OPENMP_INSTALL_LIBDIR "lib${LLVM_LIBDIR_SUFFIX}" CACHE STRING
         "Path where built OpenMP libraries should be installed.")
-  endif()
+#  endif()
 
   if (NOT MSVC)
     set(OPENMP_TEST_C_COMPILER ${LLVM_RUNTIME_OUTPUT_INTDIR}/clang)
