Index: llvm-toolchain-snapshot_20~++20241122031054+22fdc5714028/lldb/source/Plugins/Process/elf-core/ThreadElfCore.h
===================================================================
--- llvm-toolchain-snapshot_20~++20241122031054+22fdc5714028.orig/lldb/source/Plugins/Process/elf-core/ThreadElfCore.h
+++ llvm-toolchain-snapshot_20~++20241122031054+22fdc5714028/lldb/source/Plugins/Process/elf-core/ThreadElfCore.h
@@ -117,8 +117,8 @@ struct ELFLinuxSigInfo {
   static size_t GetSize(const lldb_private::ArchSpec &arch);
 };
 
-static_assert(sizeof(ELFLinuxSigInfo) == 56,
-              "sizeof ELFLinuxSigInfo is not correct!");
+//static_assert(sizeof(ELFLinuxSigInfo) == 56,
+//              "sizeof ELFLinuxSigInfo is not correct!");
 
 // PRPSINFO structure's size differs based on architecture.
 // This is the layout in the x86-64 arch case.
