Index: llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/llvm/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b.orig/llvm/CMakeLists.txt
+++ llvm-toolchain-snapshot_20~++20240920082629+ea578804c81b/llvm/CMakeLists.txt
@@ -950,7 +950,7 @@ set(LLVM_PROFDATA_FILE "" CACHE FILEPATH
 
 if(LLVM_INCLUDE_TESTS)
   # All LLVM Python files should be compatible down to this minimum version.
-  set(LLVM_MINIMUM_PYTHON_VERSION 3.8)
+  set(LLVM_MINIMUM_PYTHON_VERSION 3.5)
 else()
   # FIXME: it is unknown if this is the actual minimum bound
   set(LLVM_MINIMUM_PYTHON_VERSION 3.0)
