Index: llvm-toolchain-snapshot_20~++20241018103540+65cf7afb6d9d/mlir/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_20~++20241018103540+65cf7afb6d9d.orig/mlir/CMakeLists.txt
+++ llvm-toolchain-snapshot_20~++20241018103540+65cf7afb6d9d/mlir/CMakeLists.txt
@@ -100,6 +100,10 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
   endif()
 endif()
 
+check_symbol_exists(__mips "" __MIPS)
+append_if(__MIPS "-mxgot" CMAKE_C_FLAGS)
+append_if(__MIPS "-mxgot" CMAKE_CXX_FLAGS)
+
 # Installing the headers and docs needs to depend on generating any public
 # tablegen'd targets.
 # mlir-generic-headers are dialect-independent.
