Index: llvm-toolchain-snapshot_20~++20240909012553+b8b8fbe19dea/offload/include/Shared/APITypes.h
===================================================================
--- llvm-toolchain-snapshot_20~++20240909012553+b8b8fbe19dea.orig/offload/include/Shared/APITypes.h
+++ llvm-toolchain-snapshot_20~++20240909012553+b8b8fbe19dea/offload/include/Shared/APITypes.h
@@ -113,10 +113,10 @@ struct KernelArgsTy {
 };
 static_assert(sizeof(KernelArgsTy().Flags) == sizeof(uint64_t),
               "Invalid struct size");
-static_assert(sizeof(KernelArgsTy) ==
+/*static_assert(sizeof(KernelArgsTy) ==
                   (8 * sizeof(int32_t) + 3 * sizeof(int64_t) +
                    4 * sizeof(void **) + 2 * sizeof(int64_t *)),
-              "Invalid struct size");
+				   "Invalid struct size");*/
 
 /// Flat array of kernel launch parameters and their total size.
 struct KernelLaunchParamsTy {
