Index: llvm-toolchain-snapshot_14~++20211115123243+1e77b09538c1/compiler-rt/test/sanitizer_common/TestCases/Linux/sysconf_interceptor_bypass_test.cpp
===================================================================
--- llvm-toolchain-snapshot_14~++20211115123243+1e77b09538c1.orig/compiler-rt/test/sanitizer_common/TestCases/Linux/sysconf_interceptor_bypass_test.cpp
+++ llvm-toolchain-snapshot_14~++20211115123243+1e77b09538c1/compiler-rt/test/sanitizer_common/TestCases/Linux/sysconf_interceptor_bypass_test.cpp
@@ -1,4 +1,5 @@
 // RUN: %clangxx -O2 %s -o %t && %run %t 2>&1 | FileCheck %s
+// XFAIL: *
 
 #include <stdio.h>
 
