Index: llvm-toolchain-snapshot_17~++20230517015648+0b6264738f3d/llvm/utils/lit/lit/ProgressBar.py
===================================================================
--- llvm-toolchain-snapshot_17~++20230517015648+0b6264738f3d.orig/llvm/utils/lit/lit/ProgressBar.py
+++ llvm-toolchain-snapshot_17~++20230517015648+0b6264738f3d/llvm/utils/lit/lit/ProgressBar.py
@@ -203,15 +203,7 @@ class SimpleProgressBar:
             return
 
         for i in range(self.atIndex, next):
-            idx = i % 5
-            if idx == 0:
-                sys.stdout.write("%2d" % (i * 2))
-            elif idx == 1:
-                pass  # Skip second char
-            elif idx < 4:
-                sys.stdout.write(".")
-            else:
-                sys.stdout.write(" ")
+            sys.stdout.write('%-2d ' % (i*2))
         sys.stdout.flush()
         self.atIndex = next
 
